funnel/views/event.py

```diff
@@ -0,0 +1,177 @@
+    return render_form(form=form, title=_(u"New Event"), submit=_(u"Add Event"))
+
+
+@app.route('/<space>/event/<name>/edit', methods=['GET', 'POST'], subdomain='<profile>')
+@lastuser.requires_login
+@load_models(
+    (Profile, {'name': 'profile'}, 'g.profile'),
+    ((ProposalSpace, ProposalSpaceRedirect), {'name': 'space', 'profile': 'profile'}, 'space'),
+    (Event, {'name': 'name', 'proposal_space': 'space'}, 'event'),
+    permission='event-edit')
+def event_edit(profile, space, event):
+    form = EventForm(obj=event, model=Event)
+    if form.validate_on_submit():
+        form.populate_obj(event)
+        db.session.commit()
+        flash(_(u"Your changes have been saved"), 'info')
+        return redirect(space.url_for('admin'), code=303)
+    return render_form(form=form, title=_(u"Edit event"), submit=_(u"Save changes"))
+
+
+@app.route('/<space>/ticket_type/<name>', methods=['GET'], subdomain='<profile>')
+@lastuser.requires_login
+@load_models(
+    (Profile, {'name': 'profile'}, 'g.profile'),
+    ((ProposalSpace, ProposalSpaceRedirect), {'name': 'space', 'profile': 'profile'}, 'space'),
+    (TicketType, {'name': 'name', 'proposal_space': 'space'}, 'ticket_type'),
+    permission='ticket-type-view')
+def ticket_type(profile, space, ticket_type):
+    return render_template('ticket_type.html', profile=profile, space=space, ticket_type=ticket_type, participants=Participant.filter_by_ticket_type(ticket_type))
+
+
+@app.route('/<space>/ticket_type/new', methods=['GET', 'POST'], subdomain='<profile>')
+@lastuser.requires_login
+@load_models(
+    (Profile, {'name': 'profile'}, 'g.profile'),
+    ((ProposalSpace, ProposalSpaceRedirect), {'name': 'space', 'profile': 'profile'}, 'space'),
+    permission='new-ticket-type')
+def new_ticket_type(profile, space):
+    form = TicketTypeForm()
+    form.events.query = space.events
+    if form.validate_on_submit():
+        ticket_type = TicketType(proposal_space=space)
+        form.populate_obj(ticket_type)
+        ticket_type.make_name()
+        try:
+            db.session.add(ticket_type)
+            db.session.commit()
+        except IntegrityError:
+            db.session.rollback()
+            flash(_(u"This ticket type already exists."), 'info')
+        return redirect(space.url_for('admin'), code=303)
+    return render_form(form=form, title=_(u"New Ticket Type"), submit=_(u"Add ticket type"))
+
+
+@app.route('/<space>/ticket_type/<name>/edit', methods=['GET', 'POST'], subdomain='<profile>')
+@lastuser.requires_login
+@load_models(
+    (Profile, {'name': 'profile'}, 'g.profile'),
+    ((ProposalSpace, ProposalSpaceRedirect), {'name': 'space', 'profile': 'profile'}, 'space'),
+    (TicketType, {'name': 'name', 'proposal_space': 'space'}, 'ticket_type'),
+    permission='event-edit')
+def ticket_type_edit(profile, space, ticket_type):
+    form = TicketTypeForm(obj=ticket_type, model=TicketType)
+    form.events.query = space.events
```

Comment:
Add an `__init__` method to the form that calls `super`, then sets `self.events.query = self.obj.parent.events`.